core/translate/expr.rs

```diff
@@ -367,6 +270,7 @@ pub fn translate_condition_expr(
                             rhs: rhs_reg,
                             target_pc: jump_target_when_true,
                             flags: CmpInsFlags::default(),
+                            collation: program.curr_collation(),
```

Comment:
are these instructions ever called with anything other than `program.curr_collation()`? if not, then we don't need it, right?